packages/block/src/header.ts

```diff
@@ -178,8 +178,16 @@ export class BlockHeader {
     const nonce = toType(headerData.nonce, TypeOutput.Buffer) ?? defaults.nonce
 
     const hardforkByBlockNumber = options.hardforkByBlockNumber ?? false
-    if (hardforkByBlockNumber || options.hardforkByTTD !== undefined) {
-      this._common.setHardforkByBlockNumber(number, options.hardforkByTTD, timestamp)
+    if (
+      hardforkByBlockNumber ||
+      options.hardforkByTTD !== undefined ||
+      options.hardforkByTimestamp !== undefined
+    ) {
+      this._common.setHardforkByBlockNumber(
+        number,
+        options.hardforkByTTD,
+        options.hardforkByTimestamp ?? timestamp
```

Comment:
I'm not sure this makes sense to me.  We should just pass in `options.hardforkByTimestamp` since `timestamp` will never be undefined and this will cause the first half of [this line](https://github.com/ethereumjs/ethereumjs-monorepo/blob/109d1c9b65da5e8339136295c85aa71fedbea12d/packages/common/src/common.ts#L341) in `getHardforkByBlockNumber` to always be true, right?